### PR TITLE
Restore lost csvupload snowplow schema

### PR DESCRIPTION
**Pull request by crisptrutski**
### Description

We uploaded a version `1.0.1` for the `csvupload` events [around 5 months ago](https://metaboat.slack.com/archives/C04FZTRF457/p1698853388108969?thread_ts=1687340891.362279&cid=C04FZTRF457), but for some reason it never added or possibly removed from the repo. This meant we had a collision the [next time](https://github.com/metabase/metabase/pull/39944) we wanted to bump this schema.

This PR rewrites `1.0.1` to match what was uploaded verbatim, and rebases our latest changes to create version `1.0.2`.

### CODE REVIEW

**Review by tsmacdonald — APPROVED** [review]
:boat:

**Comment by crisptrutski** [comment]
Snowplow EE tests are failing now - thank goodness we have them. 

```
   1) CSV Uploading
       CSV Uploading (postgres)
         Cannot upload invalid.csv to a collection:
     Error: Snowplow retry timeout
      at Context.eval (../../../../../../../../../../../work/metabase/metabase/e2e/support/helpers/e2e-snowplow-helpers.js:112:12)

  2) CSV Uploading
       CSV Uploading (postgres)
         "after each" hook for "Cannot upload invalid.csv to a collection":
     AssertionError: expected [ Array(1) ] to have a length of 0 but got 1

Because this error occurred during a `after each` hook we are skipping the remaining tests in the current suite: `CSV Uploading (postgres)`
      at expectNoBadSnowplowEvents (../../../../../../../../../../../work/metabase/metabase/e2e/support/helpers/e2e-snowplow-helpers.js:94:47)
      at Context.eval (../../../../../../../../../../../work/metabase/metabase/e2e/test/scenarios/collections/uploads.cy.spec.js:124:10)

  3) CSV Uploading
       CSV Uploading (mysql)
         Cannot upload invalid.csv to a collection:
     Error: Snowplow retry timeout
      at Context.eval (../../../../../../../../../../../work/metabase/metabase/e2e/support/helpers/e2e-snowplow-helpers.js:112:12)

  4) CSV Uploading
       CSV Uploading (mysql)
         "after each" hook for "Cannot upload invalid.csv to a collection":
     AssertionError: expected [ Array(1) ] to have a length of 0 but got 1
```

Going to dig into this in the morning.

File: snowplow/iglu-client-embedded/schemas/com.metabase/csvupload/jsonschema/1-0-3 [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
```
Comment (crisptrutski) [inline snowplow/iglu-client-embedded/schemas/com.metabase/csvupload/jsonschema/1-0-3:1]: Whoops, need to delete this file before merging. 

Decided to rather change the code than update the schema yet again. A bit ick but better than having two broken schemas that never worked in our history.

**Comment by crisptrutski** [comment]
It turns out that the 1-0-1 schema published 5 months ago would never work for us, as it had "generated-columns" as required, but we did not publish it for errors. Rather than relaxing this to an optional column, I've added it to the failure message. 

Arguably this is less clean, as at this point we don't know whether we had such a side-effect, but in practice it should not matter.

**Pull request merged**